db/util_testing.go

```diff
@@ -676,8 +676,12 @@ func SetupTestDBForDataStoreWithOptions(t testing.TB, tBucket *base.TestBucket,
 	db, err := CreateDatabase(dbCtx)
 	require.NoError(t, err, "Couldn't create database 'db'")
 
-	ctx = db.AddDatabaseLogContext(ctx)
-	return db, ctx
+	return db, addDatabaseAndTestUserContext(ctx, db)
+}
+
+// addDatabaseAndTestUserContext adds a fake user to the context
+func addDatabaseAndTestUserContext(ctx context.Context, db *Database) context.Context {
+	return db.AddDatabaseLogContext(base.UserLogCtx(ctx, "gotest", base.UserDomainBuiltin, nil))
```

Comment:
Why is this not handled by the AddBucketUserLogContext?

Reply:
This is for code that works directly in a test: `collection.Put(ctx, ..)` needs a user and this provides a user. I wanted to make it clear that this wasn't a real user. This is only used by the db package.